### PR TITLE
feat: add marker-pdf reader

### PR DESCRIPTION
**Pull request by NathanLenas**
# Description

Add the [marker-pdf](https://github.com/VikParuchuri/marker/) library as a reader.
It is currently a very basic implementation.
I named the class PDFMarkerReader but it can be changed if necessary.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by Josephrp — COMMENTED** [review]
fantastic idea and pr i would like to see even more of such prs from datalab, like surya and other models !

it might be a good idea to double check licences and see if they are fully compatible for integrations such as these.

**Comment by NathanLenas** [comment]
Thank you, for your message!  You are right there might be an issue with the licensing. 
Marker is licensed under GPL-3.0, so projects using it also need to be GPL-3.0 compliant, which is not desirable for llama-index. 
However since this is a "module", maybe we can include the license in it, and only the projects using this module will have to be GPL-3.0 compliant ?

**Comment by Josephrp** [comment]
i really hope so, and look forward to the maintainers' expert guidance. 

that said there is a very good effort from the datalab & vik to solve this , so i would expect to find a solution from them , although perhaps not specifically this exact model. work is ongoing to release under favorable and more open licence terms :-)

**Comment by logan-markewich** [comment]
@NathanLenas @Josephrp I would think we can just license the integration here on its own, since it is an independent python package?

My quick google searching wasn't clear, what are the requirements of GPL-V3?

**Comment by Josephrp** [comment]
sorry say : i have absolutely no idea .

i simply love and use Vik's DataLab models for a long time and had knowledge of this issue from internal discussions. 

it could be possible that this is a non issue : as a standalone package you would simply follow the licence guidelines provided by the original licence terms, add the correct licence and mention it in the README.md

if the above non issue is indeed a non issue simply adding a licence which is the correct practice anyway would be sufficient.

with all my appologies if this may have caused a hold up unduely :-) 

i'm really looking forward to seeing datalab here :-) :-)

File: llama-index-integrations/readers/llama-index-readers-pdf-marker/README.md
Comment (Josephrp) [inline llama-index-integrations/readers/llama-index-readers-pdf-marker/README.md:28]: kindly consider : 
- adding the Vik's DataLab Surya Licence for release
- Mentioning it in the README.md
- Adding a clear explanation of the licence terms in one or two sentences in the README.md

Comment (Josephrp) [reply llama-index-integrations/readers/llama-index-readers-pdf-marker/README.md:28]: LGTM 🎉

**Review by Josephrp — APPROVED** [review]

File: llama-index-integrations/readers/llama-index-readers-pdf-marker/README.md
Comment (logan-markewich) [reply llama-index-integrations/readers/llama-index-readers-pdf-marker/README.md:28]: I'll also update the package license and add a LICENSE.md to this folder

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**
